node/bin/src/main.rs

```diff
@@ -44,14 +48,57 @@ pub async fn main() {
         }
     };
 
+    let stop_receiver_copy = stop_receiver.clone();
+
     tokio::select! {
-        _ = main_task => {},
+        _ = main_task => {
+            if *stop_receiver_copy.borrow() {
+                tracing::info!("Main task exited gracefully after stop signal");
+                // sleep to wait for other tasks to finish
+                tokio::time::sleep(GRACEFUL_SHUTDOWN_TIMEOUT).await;
```

Comment:
hmm so what exactly does this timer do? I thought components there don't listen to the `stop_receiver` - otherwise they'd exit without waiting for the timeout? 

Either way, I don't see what we are waiting for on this line...

Reply:
Ah, so it's probably applied IN CASE this main_task listens to the stop_receiver and shuts down immediately. But I think this timer doesn't help anyway because RPC is not running anymore in this case?